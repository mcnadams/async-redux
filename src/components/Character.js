import React from 'react';
import PropTypes from 'prop-types';

function Character({ character }) {
  return (
    <div>
      <img src={character.photoUrl} style={{ 'width': '100%' }}/>
      <p>{character.name}</p>
    </div>
  );
}

Character.propTypes = {
  character: PropTypes.shape({
    name: PropTypes.string.isRequired,
    photoUrl: PropTypes.string.isRequired
  })
};

export default Character;
