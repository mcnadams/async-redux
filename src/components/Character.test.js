import { shallow } from 'enzyme';
import React from 'react';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Character character={{ 
      name: 'Bonnie',
      photoUrl: './path/to/image' 
    }}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
