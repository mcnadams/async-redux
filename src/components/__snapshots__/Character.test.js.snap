// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<div>
  <img
    src="./path/to/image"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <p>
    Bonnie
  </p>
</div>
`;
